dotnetv3/S3/versioned-examples/DeleteMultipleObjectsExample/DeleteMultipleObjectsExample/DeleteMultipleObjects.cs

```diff
@@ -0,0 +1,253 @@
+            Console.WriteLine($"No. of objects successfully deleted = {errorResponse.DeletedObjects.Count}");
+            Console.WriteLine($"No. of objects failed to delete = {errorResponse.DeleteErrors.Count}");
+            Console.WriteLine("Printing error data...");
+            foreach (var deleteError in errorResponse.DeleteErrors)
+            {
+                Console.WriteLine($"Object Key: {deleteError.Key}\t{deleteError.Code}\t{deleteError.Message}");
+            }
+        }
+
+        /// <summary>
+        /// Delete multiple objects from a version-enabled bucket.
+        /// </summary>
+        /// <param name="client">The initialized S3 client object used to call
+        /// DeleteObjectVersionsAsync, DeleteObjectsAsync, and
+        /// RemoveDeleteMarkersAsync.</param>
+        /// <param name="bucketName">The name of the bucket from which to delete
+        /// objects.</param>
+        /// <param name="keys">A list of key names for the objects to delete.</param>
+        static async Task VersionedDeleteAsync(IAmazonS3 client, string bucketName, List<KeyVersion> keys)
+        {
+            var multiObjectDeleteRequest = new DeleteObjectsRequest
+            {
+                BucketName = bucketName,
+                Objects = keys, // This includes the object keys and specific version IDs.
+            };
+
+            try
+            {
+                Console.WriteLine("Executing VersionedDelete...");
+                DeleteObjectsResponse response = await client.DeleteObjectsAsync(multiObjectDeleteRequest);
+                Console.WriteLine($"Successfully deleted all the {response.DeletedObjects.Count} items");
+            }
+            catch (DeleteObjectsException ex)
+            {
+                DisplayDeletionErrors(ex);
+            }
+        }
+
+        /// <summary>
+        /// Deletes multiple objects from a non-versioned S3 bucket.
+        /// </summary>
+        /// <param name="client">The initialized S3 client object used to call
+        /// DeleteObjectVersionsAsync, DeleteObjectsAsync, and
+        /// RemoveDeleteMarkersAsync.</param>
+        /// <param name="bucketName">The name of the bucket from which to delete
+        /// objects.</param>
+        /// <param name="keys">A list of key names for the objects to delete.</param>
+        /// <returns>A list of the deleted objects.</returns>
+        static async Task<List<DeletedObject>> NonVersionedDeleteAsync(IAmazonS3 client, string bucketName, List<KeyVersion> keys)
+        {
+            // Create a request that includes only the object key names.
+            DeleteObjectsRequest multiObjectDeleteRequest = new DeleteObjectsRequest();
+            multiObjectDeleteRequest.BucketName = bucketName;
+
+            foreach (var key in keys)
+            {
+                multiObjectDeleteRequest.AddKey(key.Key);
+            }
+
+            // Execute DeleteObjects - Amazon S3 add delete marker for each object
```

Comment:
Execute DeleteObjects - Amazon S3 add delete marker for each object deletion.

(This sentence doesn't make sense to me.)

Reply:
Changed as follows:

            // Execute DeleteObjectsAsync.
            // The DeleteObjectsAsync method adds a delete marker for each
            // object deleted. You can verify that the objects were removed by
            // using the S3 console.